server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -50,7 +50,7 @@
 /**
  * Basic service for accessing remote clusters via gateway nodes
  */
-public final class RemoteClusterService extends RemoteClusterAware implements Closeable {
+public class RemoteClusterService extends RemoteClusterAware implements Closeable {
```

Comment:
I am not 100% sure of this. 

We are making a class not `final` solely to mock it in a test. Anyway to prevent this?

Reply:
Yes, I introduced a delagate method in our `Context` class and mocked that method.